tests/test_model_method.py

```diff
@@ -0,0 +1,144 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+
+from nomad.utils import get_logger
+from nomad_simulations.model_method import CoupledCluster
+
+
+def approx(value, abs=0, rel=1e-6):
+    return pytest.approx(value, abs=abs, rel=rel)
+
+
+LOGGER = get_logger(__name__)
+
+
+@pytest.mark.parametrize(
+    'name, reference',
+    [
+        (
+            'cc',
+            {
+                'exc_order': None,
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccd',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsd',
+            {
+                'exc_order': [1, 2],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsdt',
+            {
+                'exc_order': [1, 2, 3],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsdtq',
+            {
+                'exc_order': [1, 2, 3, 4],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsd(t)',
+            {
+                'exc_order': [1, 2],
+                'ptb_order': [3],
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'ccsdt(q)',
+            {
+                'exc_order': [1, 2, 3],
+                'ptb_order': [4],
+                'slv': 'variational',
+                'corr': '',
+            },
+        ),
+        (
+            'qvccd',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'quasi-variational',
+                'corr': '',
+            },
+        ),
+        (
+            'bccd',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'Brueckner',
+                'corr': '',
+            },
+        ),
+        (
+            'ccd-f12',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': 'F12',
+            },
+        ),
+        (
+            'ccd-r12',
+            {
+                'exc_order': [2],
+                'ptb_order': None,
+                'slv': 'variational',
+                'corr': 'R12',
+            },
+        ),
+    ],
+)
+def test_cc(name, reference):
```

Comment:
Add typing.

Also, you don't need to define `reference` as a dictionary, you can simply pass each quantity individually. This would help understanding the testing much more.

Also, could you be more verbose: `cc_type`, `excitation_order`, `perturbation_order`, `solver`, `explicit_correlation`? Also for the name of the testing: `test_coupled_cluster()`